stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -366,6 +389,140 @@ public struct Unsafe${Mutable}BufferPointer<Element>
   public func makeIterator() -> UnsafeBufferPointerIterator<Element> {
     return UnsafeBufferPointerIterator(_position: _position, _end: _end)
   }
+  
+  /// Deallocates the heap block previously allocated at this buffer pointer’s 
+  /// base address. 
+  ///
+  /// This buffer pointer's `baseAddress` must be `nil` or a pointer to a previously 
+  /// allocated memory block. If `baseAddress` is `nil`, this function does nothing.
+  /// Otherwise, the memory must not be initialized or `Pointee` must be a trivial 
+  /// type. This buffer pointer's `count` must be equal to the allocated 
+  /// size of the heap block.
+  @_inlineable
+  public func deallocate() {
+    _position?.deallocate()
+  }
+
+% if Mutable:
+
+  /// Allocates uninitialized memory for the specified number of instances of
+  /// type `Element`.
+  ///
+  /// The resulting buffer references a region of memory that is bound to
+  /// `Element` and is `count * MemoryLayout<Element>.stride` bytes in size.
```

Comment:
Let's add an example of allocation similar to what's in the `UnsafePointer` docs here:

```
/// The following example allocates a buffer that can store four `Int` 
/// instances and then initializes that memory with the elements of a range:
/// 
///     let buffer = UnsafeMutableBufferPointer<Int>.allocate(capacity: 4)
///     _ = buffer.initialize(from: 1...4)
///     print(buffer[2])
///     // Prints "3"
```

Reply:
done